### PR TITLE
[MIG][12.0] stock_picking_whole_scrap: Migration to v12.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT20195
cc @Tardo @carlosdauden Review please

### CODE REVIEW

File: stock_picking_whole_scrap/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+This module extends the functionality of stock module to allow scrap
```
Comment (pedrobaeza) [inline stock_picking_whole_scrap/readme/DESCRIPTION.rst]:
```suggestion
This module extends the functionality of stock module to allow to scrap
```

File: stock_picking_whole_scrap/wizards/stock_picking_whole_scrap.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    scrap_location_id = fields.Many2one(
+        comodel_name='stock.location',
+        string='Scrap Location',
+        domain=[('scrap_location', '=', True)],
+    )
+
+    @api.model
+    def default_get(self, fields):
+        if len(self.env.context.get('active_ids', list())) > 1:
+            raise UserError("You may only return one picking at a time!")
```
Comment (pedrobaeza) [inline stock_picking_whole_scrap/wizards/stock_picking_whole_scrap.py]:
```suggestion
            raise UserError("You may only scrap one picking at a time!")
```
and change translations accordingly

File: stock_picking_whole_scrap/wizards/stock_picking_whole_scrap.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            <form>
+                <group string="Detailed Operations">
+                    <field name="line_ids" nolabel="1">
+                        <tree editable="top" create="0">
+                            <field name="product_id" force_save="1"/>
+                            <field name="lot_id" force_save="1" groups="stock.group_production_lot"/>
+                            <field name="package_id" force_save="1" groups="stock.group_tracking_lot"/>
+                            <field name="owner_id" force_save="1" groups="stock.group_tracking_owner"/>
+                            <field name="quantity"/>
+                            <field name="uom_id" force_save="1" groups="product.group_uom"/>
```
Comment (pedrobaeza) [inline stock_picking_whole_scrap/wizards/stock_picking_whole_scrap.xml]:
```suggestion
                            <field name="uom_id" force_save="1" groups="uom.group_uom"/>
```

File: stock_picking_whole_scrap/tests/test_stock_picking_whole_scrap.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        cls.customer_location = cls.env.ref('stock.stock_location_customers')
+        cls.partner = cls.env['res.partner'].create({
+            'name': 'Partner - test',
+            'customer': True,
+        })
+        cls.product = cls.env['product.product'].create({
+            'name': 'test',
+            'type': 'consu',
+        })
+        cls.quant = cls.env['stock.production.lot'].create({
```
Comment (pedrobaeza) [inline stock_picking_whole_scrap/tests/test_stock_picking_whole_scrap.py]:
```suggestion
        cls.quant = cls.env['stock.quant'].create({
```
but it seems it's not being used at all or as you force `qty_done`, it's not affected.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
Changes done!

File: stock_picking_whole_scrap/wizards/stock_picking_whole_scrap.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    scrap_location_id = fields.Many2one(
+        comodel_name='stock.location',
+        string='Scrap Location',
+        domain=[('scrap_location', '=', True)],
+    )
+
+    @api.model
+    def default_get(self, fields):
+        if len(self.env.context.get('active_ids', list())) > 1:
+            raise UserError("You may only scrap one picking at a time!")
```
Comment (pedrobaeza) [inline stock_picking_whole_scrap/wizards/stock_picking_whole_scrap.py]: This is not translatable!
```suggestion
            raise UserError(_("You may only scrap one picking at a time!"))
```

**Comment by sergio-teruel** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**